cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -101,3 +151,37 @@ func (pe *ProbeEvent) initOrErrorEvent() {
 		klog.Error(err)
 	}
 }
+
+// generateUUID creates a new UUID based on the algorithm proposed in
+// https://github.com/openebs/openebs/pull/2666
+func generateUUID(bd blockdevice.BlockDevice) (string, bool) {
+	var ok bool
+	var uuidField, uuid string
+
+	// select the field which is to be used for generating UUID
+	switch {
+	case bd.DeviceAttributes.DeviceType == libudevwrapper.UDEV_PARTITION:
+		klog.Infof("device(%s) is a partition, using partition UUID: %s", bd.DevPath, bd.PartitionInfo.PartitionEntryUUID)
+		uuidField = bd.PartitionInfo.PartitionEntryUUID
+		ok = true
+	case len(bd.DeviceAttributes.WWN) > 0:
+		klog.Infof("device(%s) has a WWN, using WWN: %s", bd.DevPath, bd.DeviceAttributes.WWN)
+		uuidField = bd.DeviceAttributes.WWN
+		ok = true
+	case len(bd.PartitionInfo.PartitionTableUUID) > 0:
+		klog.Infof("device(%s) has a partition table, using partition table UUID: %s", bd.DevPath, bd.PartitionInfo.PartitionTableUUID)
+		uuidField = bd.PartitionInfo.PartitionTableUUID
+		ok = true
+	case len(bd.FSInfo.FileSystemUUID) > 0:
+		klog.Infof("device(%s) has a filesystem, using filesystem UUID: %s", bd.DevPath, bd.FSInfo.FileSystemUUID)
+		uuidField = bd.FSInfo.FileSystemUUID
+		ok = true
+	}
+
+	if ok {
+		uuid = libudevwrapper.NDMBlockDevicePrefix + util.Hash(uuidField)
```

Comment:
Can you remove the hash? I don't think it is needed with the current approach.

Reply:
One purpose of using hash was to get a common format for the UUID, irrespective of the field we use to generate the UUID.
eg: WWN can be of the format : `0x500a0751e2105248`
  FS_UUID : `d716ed29-1f2c-461a-aac7-fd7bbb835b33`
PartitionLabel : `cc5e33c5-adba-cf44-a79d-ac3c37b942a2`

using a hash will give a consistent name like `blockdevice-md5(WWN/FS_UUID/GPTLabel)`
Are there any issues with using hash @gila ?

Reply:
When I have the WWN or the UUID I can relate to that physically on the host and any other tools/SW I'm using. Those hashes are not used outside of NDM, and if I have an error on the host (showing up in the logs let say) I will see things like UUIDs and FS-IDS and not the hashes that are generated. 

Also, if given just a hash, I can not determine what the input would have been. IOW I can't go back to the WWN which makes it very unpractical IMHO.

BTW -- if a disk has a WWN, I'm pretty sure it also has a UUID, can you verify this? If that is true, then we can always show the UUID and the problem goes away?

Reply:
@gila the fields that we use to generate the UUID (i.e the value which was hashed) will be stored in `.spec.details` of the blockdevice object. So you can always associate it. The sole purpose of using the hash was to expose consistently formatted name to the users.

> BTW -- if a disk has a WWN, I'm pretty sure it also has a UUID, can you verify this?
What do you mean by the UUID here?